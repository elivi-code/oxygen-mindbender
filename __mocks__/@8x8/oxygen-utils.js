export * from '@8x8/oxygen-utils';

const useFontSize = jest.fn(() => 16);

export { useFontSize };
