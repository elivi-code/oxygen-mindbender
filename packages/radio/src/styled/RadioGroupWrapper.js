import styled from 'styled-components';
import RadioContainer from './RadioContainer';

const getFlexDirection = isHorizontal => (isHorizontal ? 'row' : 'column');

const getMargin = (isHorizontal, theme) =>
  isHorizontal
    ? `margin-left: ${theme.horizontalGroupMargin}`
    : `margin-top: ${theme.verticalGroupMargin}`;

const RadioGroupWrapper = styled.fieldset`
  border: none;
  margin: 0;
  padding: 0;
  display: flex;
  flex-direction: ${({ isHorizontal }) => getFlexDirection(isHorizontal)};
  ${RadioContainer} ~ ${RadioContainer} {
    ${props => getMargin(props.isHorizontal, props.theme)}
  }
  align-items: start;
`;

export default RadioGroupWrapper;
