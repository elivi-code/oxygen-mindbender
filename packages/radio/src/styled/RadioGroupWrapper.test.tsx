import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import { radioGroup as theme } from '@8x8/oxygen-constants';

import RadioGroupWrapper from './RadioGroupWrapper';

describe('<RadioGroupWrapper>', () => {
  let wrapper;

  function setup(props = {}) {
    return renderer
      .create(<RadioGroupWrapper theme={theme} {...props} />)
      .toJSON();
  }

  it('should style correctly the radio group if it should be displayed horizontally', () => {
    wrapper = setup({ isHorizontal: true });
    expect(wrapper).toMatchSnapshot();
  });

  it('should style correctly the radio group if it should not be displayed horizontally', () => {
    wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
