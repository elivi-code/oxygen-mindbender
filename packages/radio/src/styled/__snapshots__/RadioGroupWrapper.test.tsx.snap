// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RadioGroupWrapper> should style correctly the radio group if it should be displayed horizontally 1`] = `
.c0 {
  border: none;
  margin: 0;
  padding: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
}

<fieldset
  className="c0"
/>
`;

exports[`<RadioGroupWrapper> should style correctly the radio group if it should not be displayed horizontally 1`] = `
.c0 {
  border: none;
  margin: 0;
  padding: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
}

<fieldset
  className="c0"
/>
`;
